peerscout/keyword_extract/keyword_extract.py

```diff
@@ -144,26 +146,28 @@ def etl_keywords_get_latest_state(
             keyword_extract_config.text_field
         ]
     )
-    write_to_file(processed_data, full_file_location)
-    create_or_extend_table_schema(
-        keyword_extract_config.gcp_project,
-        keyword_extract_config.destination_dataset,
-        keyword_extract_config.destination_table,
-        full_file_location
-    )
-    write_disposition = (
-        WriteDisposition.WRITE_APPEND
-        if keyword_extract_config.table_write_append
-        else WriteDisposition.WRITE_TRUNCATE
-    )
-    load_file_into_bq(
-        filename=full_file_location,
-        table_name=keyword_extract_config.destination_table,
-        auto_detect_schema=True,
-        dataset_name=keyword_extract_config.destination_dataset,
-        write_mode=write_disposition,
-        project_name=keyword_extract_config.gcp_project
-    )
+
+    with NamedTemporaryFile() as named_temp_file:
```

Comment:
`NamedTemporaryFile` will already open the file. But you don't actually use the file object but pass the filename and open the file again.

Just checked the [docs](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile):

> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later).

That explains why it works on Unix but still feels somehow wrong. (And seems to not work on Windows)

That is why I often tend to use `TemporaryDirectory` even if I just need it for one file.

Reply:
changed to tempdirectory